src/NServiceBus.Core/Callbacks/RequestResponseExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Callbacks;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public static class RequestResponseExtensions
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="TResponse"></typeparam>
+        /// <param name="bus"></param>
+        /// <param name="requestMessage"></param>
+        /// <returns></returns>
+        public static SendContext<TResponse> RequestResponse<TResponse>(this IBus bus, object requestMessage)
+        {
+            Guard.AgainstNull(requestMessage, "requestMessage");
+            Guard.AgainstNull(bus, "bus");
+
+            var correlationId = Guid.NewGuid().ToString();
+
+            bus.SetMessageHeader(requestMessage, "NServiceBus.HasCallback", Boolean.TrueString);
```

Comment:
>Sure, what should I call it ?

I suck at names but is `NServiceBus.Callbacks` to vague?

On Mon, Mar 23, 2015 at 8:35 AM, Andreas Öhlund <
andreas.ohlund@particular.net> wrote:

> Since RequestResponse is a pattern that is used when users do
> bus.Send/bus.Reply (async) I propose we rename this to `bus.SyncronousRequestResponse`
> to make clear that there is a difference?
>
> On Mon, Mar 23, 2015 at 6:54 AM, John Simons <notifications@github.com>
> wrote:
>
>> In src/NServiceBus.Core/Callbacks/RequestResponseExtensions.cs
>> <https://github.com/Particular/NServiceBus/pull/2689#discussion_r26915353>
>> :
>>
>> > +    {
>> > +        /// <summary>
>> > +        ///
>> > +        /// </summary>
>> > +        /// <typeparam name="TResponse"></typeparam>
>> > +        /// <param name="bus"></param>
>> > +        /// <param name="requestMessage"></param>
>> > +        /// <returns></returns>
>> > +        public static SendContext<TResponse> RequestResponse<TResponse>(this IBus bus, object requestMessage)
>> > +        {
>> > +            Guard.AgainstNull(requestMessage, "requestMessage");
>> > +            Guard.AgainstNull(bus, "bus");
>> > +
>> > +            var correlationId = Guid.NewGuid().ToString();
>> > +
>> > +            bus.SetMessageHeader(requestMessage, "NServiceBus.HasCallback", Boolean.TrueString);
>>
>> I think is more
>> NServiceBus.ThisHeaderIsToSupportAFutureVersionOfRoutingThatWeHaventReleasedYet
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/Particular/NServiceBus/pull/2689/files#r26915353>.
>>
>
>